conformance/tests/httproute-service-types.go

```diff
@@ -0,0 +1,183 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"slices"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	discoveryv1 "k8s.io/api/discovery/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+	client "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteServiceTypes)
+}
+
+var HTTPRouteServiceTypes = suite.ConformanceTest{
+	ShortName:   "HTTPRouteServiceTypes",
+	Description: "A single HTTPRoute should be able to route traffic to various service type backends",
+	Features: []suite.SupportedFeature{
+		suite.SupportGateway,
+		suite.SupportHTTPRoute,
+	},
+	Manifests: []string{"tests/httproute-service-types.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		var (
+			typeManualEndpoints = []string{
+				"manual-endpoints",
+				"headless-manual-endpoints",
+			}
+
+			typeManualEndpointSlices = []string{
+				"manual-endpointslices",
+				"headless-manual-endpointslices",
+			}
+
+			typeManaged = []string{
+				"headless",
+			}
+
+			serviceTypes = slices.Concat(typeManaged, typeManualEndpoints, typeManualEndpointSlices)
```

Comment:
i think we'll need to bump the go.mod go minimum version to 1.22 since this is a 1.22 feature

Reply:
good catch